### PR TITLE
wifi: try to connect to known network before listing

### PR DESCRIPTION
**Pull request by avsej**
Iterate over available networks ordered by signal power, and check if we
can connect to saved SSIDs. If we can, then skip network list dialog.

### CODE REVIEW

File: frontend/ui/network/manager.lua [hunk 1]
```diff
@@ -273,10 +273,29 @@ function NetworkMgr:showNetworkMenu(complete_callback)
...
+        table.foreach(network_list,
+           function(idx, network)
+               if network.password then
+                   success = NetworkMgr:authenticateNetwork(network)
+                   if success then
+                       NetworkMgr:obtainIP()
+                       if complete_callback then
+                           complete_callback()
+                       end
+                       UIManager:show(InfoMessage:new{text = _("Connected."), timeout = 3})
```
[the remark this answers is not in the document]
Comment (avsej) [reply frontend/ui/network/manager.lua]: The existing popup does not mention the network name:
https://github.com/koreader/koreader/blob/master/frontend/ui/widget/networksetting.lua#L218-L228

**Comment by Frenzie** [comment]
What device are you on? That sounds like a bug.

File: frontend/ui/network/manager.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (Frenzie) [reply frontend/ui/network/manager.lua]: I don't know in what scenario that would show. I'm talking about the regular "Connected to network *insert network name*" message.

**Comment by avsej** [comment]
cervantes 4, why koreader would need to show me the scan results before trying to connect?

File: frontend/ui/network/manager.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (avsej) [reply frontend/ui/network/manager.lua]: @Frenzie, I've fixed in current patch

**Comment by pazos** [comment]
Related to #4422 

Kobos, Cervantes and Sony devices are affected by this PR.  Unlike kobos, sony/cervantes devices don't repurpose a wpasupplicant configuration, so this is a way to reconnect to a known ap without going through the network menu and provides a way to automate this process after suspension.

It needs testing on kobos, or just create a new function like `reconnectOrShowNetworkMenu` and use that in [cervantes](https://github.com/koreader/koreader/blob/master/frontend/device/cervantes/device.lua#L163) and [sony](https://github.com/koreader/koreader/blob/master/frontend/device/sony-prstux/device.lua#L138) turnOnWifi implementations.

**Comment by Frenzie** [comment]
> just create a new function like reconnectOrShowNetworkMenu and use that in cervantes and sony turnOnWifi implementations.

Work for me.

**Comment by avsej** [comment]
i've implemented `reconnectOrShowNetworkMenu`, but added only for cervantes as I don't have sony device to test

**Comment by Frenzie** [comment]
@pazos Any comments before merging?

**Comment by pazos** [comment]
@Frenzie: nope, lgtm

**Comment by Frenzie** [comment]
Thanks @avsej !

**Pull request merged**